src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,57 @@
+type Props = {
+  query: string;
+  setQuery: (value: string) => void;
+  selectFilter: string;
+  setSelectFilter: (option: string) => void;
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  setQuery,
+  selectFilter,
+  setSelectFilter,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectFilter}
+            onChange={e => setSelectFilter(e.target.value)}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={e => setQuery(e.target.value)}
```

Comment:
Similarly, add a type annotation for the event parameter in the `onChange` handler for the input element. Use `React.ChangeEvent<HTMLInputElement>` to ensure type safety.